mitiq/benchmarks/mirror_circuits.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+# coding: utf-8
+# %%
+
+# %%
+
+
+import cirq
+import numpy as np
+from numpy import random
+import networkx as nx
+from typing import Optional
+from cirq.experiments.qubit_characterizations import _single_qubit_cliffords
+
+
+# %%
+paulis = [cirq.X, cirq.Y, cirq.Z, cirq.I]
+
+def random_paulis(nqubits: int) -> cirq.Circuit:
+    """Returns a circuit with a random pauli gate applied to each qubit
+
+    Args:
+    nqubits: The number of qubits in the circuit
+    """
+    circuit = cirq.Circuit()
+    circuit.append((paulis[random.randint(4)](cirq.LineQubit(x)) for x in range(nqubits)))
+    return circuit
+
+
+# %%
+
+
+def edge_grab(xi: float, connectivity_graph_: nx.Graph) -> nx.Graph:
+    """Returns a set of edges for which two qubit gates are to be applied given a two qubit gate density, xi,
+    and the connectivity graph that must be satisfied
+
+    Args:
+        xi: Two qubit gate density
+        connectivity_graph_: The connectivity graph for the backend on which the circuit will be run
+    """
+    connectivity_graph = nx.Graph()
+    connectivity_graph.add_nodes_from(connectivity_graph_)
+    connectivity_graph.add_edges_from(connectivity_graph_.edges)
+    candidate_edges = nx.Graph()
+    
+    nqubits = connectivity_graph.number_of_nodes()
+
+    while(len(list(connectivity_graph.edges))>0):
```

Comment:
Do `while connectivity_graph.edges:` for this loop. It's shorter and more Pythonic.